src/js/intlTelInput.js

```diff
@@ -1063,7 +1058,8 @@ Plugin.prototype = {
   _beforeSetNumber: function(number) {
     if (this.options.separateDialCode) {
       var dialCode = this._getDialCode(number);
-      if (dialCode) {
+      // _getDialCode returns empty string if there is no dialCode
+      if (dialCode && dialCode !== "") {
```

Comment:
Oops, sure, I was thinking in strict equality... Will remove it now for sake of beauty!